analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotCallGCSuppressFinalize.cs

```diff
@@ -56,12 +56,8 @@ public sealed class DoNotCallGCSuppressFinalize : DoNotCallMethodsCSharpBase
             }
 
             var methodSymbol = semanticModel.GetDeclaredSymbol(methodDeclaration);
-            if (!methodSymbol.IsIDisposableDispose())
-            {
-                return true;
-            }
-
-            return false;
+            return !methodSymbol.IsIDisposableDispose()
```

Comment:
inside `SonarAnalyzer.Common\Rules\DoNotCallMethodsBase.cs`, `ShouldReportOnMethodCall` is called at the end of the method, after using the semantic model multiple times

could you add an AST-based name check on the `methodDeclaration` (`Dispose` or `DisposeAsync`) in the override of `IsInValidContext` which is called at the very beginning? like this, you'd save a lot of semantic model calls from the starts

Reply:
actually, my comment is wrong

however, we can still improve the implementation of `DoNotCallMethodsBase` by moving some name checks earlier, and like this we'd improve 3 different rules easily.

Reply:
I opened https://github.com/SonarSource/sonar-dotnet/issues/3854